scrapers/StateOfTexas.mjs

```diff
@@ -3,30 +3,42 @@ import moment from 'moment';
 
 export default ({ database, DataScraper }) => {
 	return DataScraper(database, 'StateOfTexas', async () => {
-		const data = [];
+
+		// Definitions
+		const data=[]
 		const baseUrl = 'https://api.tnris.org/api/v1';
-		const id = '2679b514-bb7b-409f-97f3-ee3879f34448';
-		const collections = `collections/${id}`;
-		const resources = `resources?collection_id=${id}`;
-		const collectionsRequest = await fetch(`${baseUrl}/${collections}`);
-		const collection = await collectionsRequest.json();
-		const resourcesRequest = await fetch(`${baseUrl}/${resources}`);
-		const resource = await resourcesRequest.json();
 
-		const { publication_date, description, name } = collection;
-		const { results } = resource;
-		const created = moment(publication_date, 'YYYY-MM-DD').unix();
+		// Getting collection ID data
+		const id_data = await fetch('https://api.tnris.org/api/v1/collections_catalog/')
+							  .then(response => response.json())
+		const idArray = id_data.results.map(function(item) { return item["collection_id"]; });
+		// Looping over each collection_id
+		for (const id of idArray) {
+			var collections = `collections/${id}`
+			var resources = `resources?collection_id=${id}`
+			var collectionsRequest = await fetch(`${baseUrl}/${collections}`)
+			var collection = await collectionsRequest.json()
+			var resourcesRequest = await fetch(`${baseUrl}/${resources}`)
+			var resource = await resourcesRequest.json()
 
-		results.map(item => {
-			return data.push({
-				url: item.resource,
-				updated: 0,
-				created,
-				description,
-				name,
-			});
-		});
+			var { publication_date, description, name } = collection;
+			var { results } = resource;
+			var created = moment(publication_date, 'YYYY-MM-DD').unix();
+			
+			// Only adding results with content to the data array
+			if( results.length>0){
 
+					results.map(item => {
```

Comment:
Even though this was a mistake in the original code, the array `map` function should not be used if you're not interested in the return value. Use `forEach` or better yet, `for (const val of vals)`.